java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java

```diff
@@ -145,37 +188,79 @@ public TransferPair getTransferPair(String ref, BufferAllocator allocator, CallB
     return new NullableMapTransferPair(this, new MapVector(ref, allocator, fieldType, callBack), false);
   }
 
-  protected class NullableMapTransferPair extends MapTransferPair {
+  protected class NullableMapTransferPair implements TransferPair {
```

Comment:
`NullableMapTransferPair` -> `MapTransferPair`?

Reply:
I think this would be ok to rename since its a protected inner class